### PR TITLE
fix: user management api to groups

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/UserManagement/PermissionUtils.res [hunk 1]
```diff
@@ -1,224 +1,85 @@
...
+    workflowsView: getAccessValue(~permissionValue=WorkflowsView, ~permissionList),
+    workflowsManage: getAccessValue(~permissionValue=WorkflowsManage, ~permissionList),
+    analyticsView: getAccessValue(~permissionValue=AnalyticsView, ~permissionList),
+    usersView: getAccessValue(~permissionValue=UsersView, ~permissionList),
+    usersManage: getAccessValue(~permissionValue=UsersManage, ~permissionList),
+    merchantDetailsView: getAccessValue(~permissionValue=MerchantDetailsView, ~permissionList),
+    merchantDetailsManage: getAccessValue(~permissionValue=MerchantDetailsManage, ~permissionList),
+    organizationManage: getAccessValue(~permissionValue=OrganizationManage, ~permissionList),
   }
+  Js.log2("abcabcabc", abc)
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/PermissionUtils.res:81]: remove this log

File: src/screens/UserManagement/PermissionUtils.res [hunk 2]
```diff
@@ -1,224 +1,85 @@
...
-      ~permissionValue=SurchargeDecisionManagerWrite,
-      ~permissionList,
-    ),
-    surchargeDecisionManagerRead: getAccessValue(
-      ~permissionValue=SurchargeDecisionManagerRead,
-      ~permissionList,
-    ),
-    usersRead: getAccessValue(~permissionValue=UsersRead, ~permissionList),
-    usersWrite: getAccessValue(~permissionValue=UsersWrite, ~permissionList),
+  Js.log2("permissionListpermissionList-111111", permissionList)
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/PermissionUtils.res:66]: remove log

File: src/entryPoints/Provider/ProviderTypes.res [hunk 3]
```diff
@@ -25,6 +25,7 @@ type getInfoType = {
   module_: string,
   description: string,
   mutable permissions: array<permissions>,
+  mutable isPermissionAllowed: bool,
```
Comment (JeevaRamu0104) [inline src/entryPoints/Provider/ProviderTypes.res:28]: why this is used?

File: src/screens/UserManagement/PermissionUtils.res [hunk 4]
```diff
@@ -1,224 +1,85 @@
...
+  let abc = {
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/PermissionUtils.res:67]: rename the variable name

File: src/entryPoints/Provider/ProviderTypes.res (hunk 3 above)
Comment (JeevaRamu0104) [reply src/entryPoints/Provider/ProviderTypes.res:28]: `isPermissionAllowed` inside type permissions is not required

File: src/screens/UserManagement/UserManagementTypes.res [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  merchantDetailsView: authorization,
+  merchantDetailsManage: authorization,
+  organizationManage: authorization,
+}
+
+// TODO : Remove this type definition as no longer used
+type permissions = {
+  description: string,
+  enum_name: string,
+  mutable isPermissionAllowed: bool,
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserManagementTypes.res:36]: remove isPermissionAllowed type definition it not required

Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserManagementTypes.res:36]: Will refactor and remove unused type definitions in the next PR .

File: src/screens/UserManagement/PermissionUtils.res [hunk 6]
```diff
@@ -1,223 +1,81 @@
...
+  | "merchant_details_manage" => MerchantDetailsManage
+  | "organization_manage" => OrganizationManage
   | val => UnknownPermission(val)
   }
 }
 
 let getAccessValue = (~permissionValue: permissionType, ~permissionList) => {
   let isPermissionFound = permissionList->Array.find(ele => {
-    ele === permissionValue
+    Js.log4("permissionListpermissionList", permissionList, permissionValue, ele == permissionValue)
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/PermissionUtils.res:42]: please remove this log

File: src/screens/UserManagement/UserManagementUtils.res [hunk 7]
```diff
@@ -81,33 +81,29 @@ let roleType = roleListData =>
...
+    if copyOfPermissionsData->Array.includes(infoValItem.module_) {
+      infoValItem.isPermissionAllowed = true
+    } else {
+      infoValItem.isPermissionAllowed = false
+    }
     infoValItem
   })
 }
 
 let defaultPresentInInfoList = infoData => {
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserManagementUtils.res:102]: this function is not required if we remove` src/screens/UserManagement/UserManagementTypes.res` type permission

File: src/screens/UserManagement/UserManagementTypes.res (hunk 5 above)
Comment (JeevaRamu0104) [reply src/screens/UserManagement/UserManagementTypes.res:36]: https://github.com/juspay/hyperswitch-control-center/pull/471#discussion_r1510867960

File: src/screens/UserManagement/UserManagementUtils.res (hunk 7 above)
Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserManagementUtils.res:102]: This `isPermissionAllowed` is for module level permissions

Comment (JeevaRamu0104) [reply src/screens/UserManagement/UserManagementUtils.res:102]: https://github.com/juspay/hyperswitch-control-center/pull/471#discussion_r1510855022

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**